core/src/main/java/com/sap/ai/sdk/core/AiCoreServiceWithDeployment.java

```diff
@@ -0,0 +1,156 @@
+package com.sap.ai.sdk.core;
+
+import com.sap.ai.sdk.core.client.DeploymentApi;
+import com.sap.ai.sdk.core.client.model.AiDeployment;
+import com.sap.cloud.sdk.cloudplatform.connectivity.DefaultHttpDestination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.HttpDestination;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+
+/** Connectivity convenience methods for AI Core with deployment. */
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class AiCoreServiceWithDeployment implements AiCoreServiceStub {
+
+  // the deployment id to be used
+  @Nonnull private final Function<AiCoreServiceWithDeployment, String> deploymentId;
+
+  // the base destination to be used
+  @Nonnull private final Supplier<Destination> destination;
+
+  // the resource group, "default" if null
+  @Nullable private final String resourceGroup;
+
+  /**
+   * Create a new instance of the AI Core service with a specific deployment id and destination.
+   *
+   * @param deploymentId The deployment id handler.
+   * @param destination The destination handler.
+   */
+  public AiCoreServiceWithDeployment(
+      @Nonnull final Function<AiCoreServiceWithDeployment, String> deploymentId,
+      @Nonnull final Supplier<Destination> destination) {
+    this(deploymentId, destination, null);
```

Comment:
```suggestion
  @Nonnull private final String resourceGroup;

  /**
   * Create a new instance of the AI Core service with a specific deployment id and destination.
   *
   * @param deploymentId The deployment id handler.
   * @param destination The destination handler.
   */
  public AiCoreServiceWithDeployment(
      @Nonnull final Function<AiCoreServiceWithDeployment, String> deploymentId,
      @Nonnull final Supplier<Destination> destination) {
    this(deploymentId, destination, "default");
```

Nullable should be avoided